src/renderers/shared/reconciler/ReactCompositeComponent.js

```diff
@@ -122,10 +124,18 @@ var ReactCompositeComponentMixin = {
    * @final
    * @internal
    */
-  mountComponent: function(rootID, transaction, context) {
+  mountComponent: function(
+    rootID,
+    transaction,
+    nativeParent,
+    nativeContainerInfo,
+    context
+  ) {
     this._context = context;
     this._mountOrder = nextMountID++;
     this._rootNodeID = rootID;
+    this._nativeParent = nativeParent;
+    this._nativeContainerInfo = nativeContainerInfo;
```

Comment:
Seems like a waste to store it on everything just to solve the root case. Maybe we need to refactor the root? The top level wrapper seems a little hacky too. That could be a follow up though.